static/app/views/alerts/rules/metric/ruleConditionsForm.tsx

```diff
@@ -99,15 +121,22 @@ type Props = {
 
 type State = {
   environments: Environment[] | null;
+  filterKeys: TagCollection;
+  measurements: MeasurementCollection;
 };
 
 class RuleConditionsForm extends PureComponent<Props, State> {
   state: State = {
     environments: null,
+    measurements: {},
+    filterKeys: {},
   };
 
   componentDidMount() {
     this.fetchData();
+    const measurements = getMeasurements();
+    const filterKeys = this.getFilterKeys();
+    this.setState({measurements, filterKeys});
```

Comment:
Do these need to be in state? Are you doing this to memoize them?

Reply:
Ya pretty much

Reply:
Okay so I do like that `filterKeys` is memoized, but it looks like this is preventing custom tags from being shown. `this.getFilterKeys` depends on `tags` which are passed in as a prop and those can change depending on what project is displayed. So I think you'll need to make sure that this reacts to changes in `tags`